### ISSUE TITLE
Request failed: Cloudflare reCaptcha detected

### ISSUE THREAD

**Issue by buccinator** [SickChill/SickChill#5952]
### Before submitting your issue:

Enable debug logging in SickChill settings, reproduce the error (be sure to disable after the bug is fixed)

Branch/Commit: develop
OS: Ubuntu 18.04.4 LTS

Browser: N/A

What you did:
recently moved to tor + polipo (primarily because my stupid country blocked opensubtitles.org) so i guess with doing it this way the issue was expected. i also tried "pip install cloudscraper --upgrade" and "pip install cfscrape --upgrade"

What happened:
sickchill started encountering reCaptchas and failing to get past them with either libs/cfscrape or libs/cloudscraper

What you expected:
tried searching for an answer with no luck, i assume its not unique to me.

Logs:
```
2020-02-03 10:41:14 INFO     SEARCHQUEUE-DAILY-SEARCH :: Beginning daily search for new episodes
2020-02-03 10:41:32 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [LimeTorrents] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:41:35 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [MagnetDL] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:41:47 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [SceneTime] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:41:47 WARNING  SEARCHQUEUE-DAILY-SEARCH :: [SceneTime] :: Unable to connect to provider
2020-02-03 10:42:04 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [SkyTorrents] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:42:13 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [ThePirateBay] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:42:18 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [ThePirateBay] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:42:22 ERROR    SEARCHQUEUE-DAILY-SEARCH :: [Torrentz] :: [4b7f06d] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter.
2020-02-03 10:42:24 INFO     SEARCHQUEUE-DAILY-SEARCH :: No needed episodes found
```

**Comment by miigotu** [comment]
This is a proxying issue, you can't use a VPN instead?

**Comment by buccinator** [comment]
isnt it more a tor issue than proxying issue? but thats not to say that a normal user wouldnt encounter it anyway just far less frequently. a no i cant use a VPN, i have sickchill installed on a server with other services and a VPN would impact on them. cloudscrape is installed with sickchill and afaik just isnt working as it should

**Comment by miigotu** [comment]
Cloudscraper/cfscrape do not solve captchas, just challenge/response. cloudscraper supports using them, but we have not implemented use of them in SC yet. 
Cloudflare will throw up a captcha on every request from a tor endpoint.

@VeNoMouS this is due to requiring use of 3rd party captcha solver correct?

**Comment by buccinator** [comment]
@miigotu ah ok makes sense now. it gets a new ip every 20mins anyway and not every scraper is faces a captcha each time, just whatever cloudflare hosted site at the time that deems the current exit node as a threat. its bizarre how cloudflare can see the exit node ip as a threat on one site but not another. guess ill just have to stick with hoping one of the scrapes lands.. eventually. better chance than it never landing at all anymore anyway :p

**Comment by miigotu** [comment]
Cloudflare throws a captcha on 94% of exit nodes, but not every site has cloudflare ^

**Comment by miigotu** [comment]
if you are interested in adding support for a 3rd party captcha solver, let me know.

https://github.com/venomous/cloudscraper#3rd-party-recaptcha-solvers

**Comment by buccinator** [comment]
> Cloudflare throws a captcha on 94% of exit nodes, but not every site has cloudflare ^

i understand that but i am talking about the sites which do but not blacklist the same ip, cloudflare must still base the rules by site. nfi.

**Comment by buccinator** [comment]
> if you are interested in adding support for a 3rd party captcha solver, let me know.
> 
> https://github.com/venomous/cloudscraper#3rd-party-recaptcha-solvers

if paying for that service is cheaper than paying for a http or socks5 proxy, then sure! thanks again for your feedback

**Comment by VeNoMouS** [comment]
tbh, that many sites should'nt triggering reCaptcha some of those sites arent even behind CF, i think something else is going on?!

**Comment by BenjV** [comment]
Use a VPN and only redirect SickChill to the VPN tunnel and not the other applications on the server.

**Comment by miigotu** [comment]
@buccinator It looks like installing brotli might help from his code.
```python
    def Challenge_Response(self, resp, **kwargs):
        if self.is_reCaptcha_Challenge(resp):
            # ------------------------------------------------------------------------------- #
            # double down on the request as some websites are only checking
            # if cfuid is populated before issuing reCaptcha.
            # ------------------------------------------------------------------------------- #

            resp = self.decodeBrotli(
                super(CloudScraper, self).request(resp.request.method, resp.url, **kwargs)
            )

            if not self.is_reCaptcha_Challenge(resp):
                return resp

            # ------------------------------------------------------------------------------- #
            # if no reCaptcha provider raise a runtime error.
            # ------------------------------------------------------------------------------- #

            if not self.recaptcha or not isinstance(self.recaptcha, dict) or not self.recaptcha.get('provider'):
                sys.tracebacklimit = 0
                raise RuntimeError(
                    "Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider "
                    "correctly via the 'recaptcha' parameter."
                )
```

@VeNoMouS this is in cloudscraper, I'm confused why its hitting it too. I feel like his proxy isnt passing everything through or is using something from cloudflare in the chain.

@BenjV thats what I would do too, but it would be nice to track this down and maybe support the 3rd party captcha providers in SC

**Comment by VeNoMouS** [comment]
1. **reCaptcha**
It seems really odd... i would suggest he try access those sites with debug enabled so we can get a more complete picture of what is occurring... because in order for it to trigger for reCaptcha it has to hit a couple key requires..

1.1. Response "server" starts with "cloudflare"
1.2. Status code is 403
1.3. `__cf_chl_captcha_tk__` is in the response body

2. ** brotli**
... this is covered in the function below here... the response  header `Content-Encoding` has to be set to `br`, there are safety checks in place to detect this, the body will come out gibberish otherwise...

```python

    def decodeBrotli(self, resp):
        if requests.packages.urllib3.__version__ < '1.25.1' and resp.headers.get('Content-Encoding') == 'br':
            if self.allow_brotli and resp._content:
                resp._content = brotli.decompress(resp.content)
            else:
                logging.warning(
                    'You\'re running urllib3 {}, Brotli content detected, '
                    'Which requires manual decompression, '
                    'But option allow_brotli is set to False, '
                    'We will not continue to decompress.'.format(requests.packages.urllib3.__version__)
                )

        return resp
```

**Comment by buccinator** [comment]
> 1. It seems really odd... i would suggest he try access those sites with debug enabled so we can get a more complete picture of what is occurring... because in order for it to trigger for reCaptcha it has to hit a couple key requires..

roger that, will do so again tonight with debug turned on

**Comment by VeNoMouS** [comment]
do something more along the lines of 

```python
import cloudscraper
scraper = cloudscraper.create_scraper(debug=True)
scraper.get('https://www.yoururl/')
```

it will print out the whole request headers and body..

**Comment by buccinator** [comment]
i did previously also got the error "RuntimeError: !!Loop Protection!! We have tried to solve 3 time(s) in a row." but now it seems the reCaptcha message is gone and replaced with 403.

i turned on debug in SC and with /opt/sickchill/lib/cloudscraper/\_\_init\_\_.py starting @ line 82, changed (what i think you were asking me to do below) and compiled it

`class CloudScraper(Session):`

`    def __init__(self, *args, **kwargs):`
`#        self.debug = kwargs.pop('debug', False)`
`        self.debug = kwargs.pop('debug', True)`
`        self.delay = kwargs.pop('delay', None)`
`        self.cipherSuite = kwargs.pop('cipherSuite', None)`
`        self.interpreter = kwargs.pop('interpreter', 'native')`
`#       self.recaptcha = kwargs.pop('recaptcha', {})`
`        self.recaptcha = kwargs.pop('recaptcha', {'provider': 'return_response'})`

`        self.allow_brotli = kwargs.pop(`
`            'allow_brotli',`
`            True if 'brotli' in sys.modules.keys() else False`
`        )`

>2020-02-04 18:47:11 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: Fallback season pack search initiate
2020-02-04 18:47:11 INFO     SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: Performing season pack search for American Masters
2020-02-04 18:47:11 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: Search Mode: Season
2020-02-04 18:47:11 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: Search string: American Masters S01
2020-02-04 18:47:14 INFO     SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: Request failed: 403 Client Error: Forbidden for url: http://www.magnetdl.com/a/american-masters-s01/ (<type 'str'>)
2020-02-04 18:47:14 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: URL did not return results/data, if the results are on the site maybe try a custom url, or a different one
2020-02-04 18:47:14 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Performing episode search for American Masters
2020-02-04 18:47:26 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Request failed: 403 Client Error: Forbidden for url: https://www.scenetime.com/takelogin.php?__cf_chl_jschl_tk__=4172b2e17dc5413ee930ffba80d0fa7292166e80-1580802436-0-Aby9VsMXvM-UIyP3paRBybjRNa6CqAHOp1uSbTkI5JB-rWKZb-P4X4JfpFKEY6TpXZhHzofuTppDwCDbpbPTRDnQie085O8hKD6G8jiJd1cBjaRJOqWHSax5sRchU3UQ_WtViopm0hxL84w0Om7xEJW_eoE1yzvSpaLzpnYzwqpJ2IJYArtvhJAonIHzOQ078rEpAT6ztblobvBf0z5RPFeQOGaq-Ts65bW9Z35mvbFludNp21ONuO9jbmIw0l976UUYXxaoALIkoJau3t_F9EwwIZwR4OfuosFBlHxELLJnavwdeKeoi4jWym-w56649BK3u-54bCtNnrRoJDYmQbRILPXfbu4Mn3mVJ2cLIvN9 (<type 'str'>)
2020-02-04 18:47:26 WARNING  SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Unable to connect to provider
2020-02-04 18:47:26 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode
2020-02-04 18:47:26 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search string: American Masters S01E02
2020-02-04 18:47:31 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Request failed: 403 Client Error: Forbidden for url: https://www.scenetime.com/browse.php (<type 'str'>)
2020-02-04 18:47:31 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode
2020-02-04 18:47:31 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search string: American Masters S01E03
2020-02-04 18:47:33 ERROR    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: [e388002] Request failed: !!Loop Protection!! We have tried to solve 3 time(s) in a row. (<type 'str'>)
Traceback (most recent call last):
RuntimeError: !!Loop Protection!! We have tried to solve 3 time(s) in a row.
2020-02-04 18:47:33 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Traceback (most recent call last):
RuntimeError: !!Loop Protection!! We have tried to solve 3 time(s) in a row.

>2020-02-04 18:47:34 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode
2020-02-04 18:47:34 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search string: American Masters S01E04
2020-02-04 18:47:38 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Request failed: 403 Client Error: Forbidden for url: https://www.scenetime.com/browse.php (<type 'str'>)
2020-02-04 18:47:38 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode
2020-02-04 18:47:38 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search string: American Masters S01E05
2020-02-04 18:47:43 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Request failed: 403 Client Error: Forbidden for url: https://www.scenetime.com/browse.php (<type 'str'>)
2020-02-04 18:47:43 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode
2020-02-04 18:47:43 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search string: American Masters S01E06
2020-02-04 18:47:47 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Request failed: 403 Client Error: Forbidden for url: https://www.scenetime.com/browse.php (<type 'str'>)
2020-02-04 18:47:48 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode
2020-02-04 18:47:48 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search string: American Masters S01E07

>2020-02-04 18:50:43 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: Search Mode: Episode
2020-02-04 18:50:43 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: Search string: American Masters S01E06
2020-02-04 18:50:48 INFO     SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: Request failed: 403 Client Error: Forbidden for url: https://thepiratebay.org/search/American%20Masters%20S01E06/0/7/200 (<type 'str'>)
2020-02-04 18:50:48 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: URL did not return data, maybe try a custom url, or a different one
2020-02-04 18:50:48 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: Search Mode: Episode
2020-02-04 18:50:48 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: Search string: American Masters S01E07
2020-02-04 18:51:18 INFO     SEARCHQUEUE-BACKLOG- :: Beginning backlog search for: [American Masters]
2020-02-04 18:51:18 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Last update was too soon, using old cache: 2020-02-04 18:43:08. Updated less then 10 minutes ago
2020-02-04 18:51:18 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [LimeTorrents] :: Last update was too soon, using old cache: 2020-02-04 18:43:03. Updated less then 10 minutes ago
2020-02-04 18:51:18 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [TokyoToshokan] :: Last update was too soon, using old cache: 2020-02-04 18:45:59. Updated less then 15 minutes ago
2020-02-04 18:51:18 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [ThePirateBay] :: Last update was too soon, using old cache: 2020-02-04 18:35:19. Updated less then 30 minutes ago
2020-02-04 18:51:18 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SkyTorrents] :: Search Mode: RSS
2020-02-04 18:51:22 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SkyTorrents] :: Request failed: 403 Client Error: Forbidden for url: https://www.skytorrents.lol/top100?category=show&sort=created&type=video&category=show&sort=created&type=video (<type 'str'>)
2020-02-04 18:51:22 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SkyTorrents] :: Data returned from provider does not contain any torrents
2020-02-04 18:51:22 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [Torrentz] :: Last update was too soon, using old cache: 2020-02-04 18:45:59. Updated less then 15 minutes ago
2020-02-04 18:51:22 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [Nyaa] :: Search Mode: RSS
2020-02-04 18:51:24 INFO     SEARCHQUEUE-BACKLOG-81727 :: [Nyaa] :: Request failed: 403 Client Error: Forbidden for url: https://nyaa.si/?s=id&c=1_0&page=rss&o=desc&f=2 (<type 'str'>)
2020-02-04 18:51:24 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [Nyaa] :: RSS error:
2020-02-04 18:51:24 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [Nyaa] :: Data returned from provider does not contain any torrents
2020-02-04 18:51:24 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [HorribleSubs] :: Last update was too soon, using old cache: 2020-02-04 18:45:55. Updated less then 15 minutes ago
2020-02-04 18:51:24 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [Rarbg] :: Last update was too soon, using old cache: 2020-02-04 18:43:11. Updated less then 10 minutes ago
2020-02-04 18:51:24 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [MagnetDL] :: Last update was too soon, using old cache: 2020-02-04 18:43:05. Updated less then 10 minutes ago
2020-02-04 18:51:24 INFO     SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Performing episode search for American Masters
2020-02-04 18:51:24 DEBUG    SEARCHQUEUE-BACKLOG-81727 :: [SceneTime] :: Search Mode: Episode


i have no idea what im doing with python if that isnt obvious. if it were perl, php, java or c# id have a better chance in helping. hell, took me ages to figure out how to even post this so it didnt look like dog shit.

**Comment by VeNoMouS** [comment]
from yoru paste... i can see sceneTime is behind cloudflare... and does have reCaptcha

![image](https://user-images.githubusercontent.com/597545/73726752-f63ac500-4794-11ea-80a2-cc34a99f1dd0.png)

where as earlier your output was that every provider had reCaptcha... could be weird bug from sickchill not detecting the raised exception.... anyway... 100% that would need to be reCaptcha solved before you could continue... @miigotu would need to add support for pass reCaptcha options for **cloudscraper**..

**Comment by VeNoMouS** [comment]
FWIW,  skytorrrents is also behind cloudflare...

**Comment by VeNoMouS** [comment]
Question... are you using tor exit nodes?

**Comment by buccinator** [comment]
> from yoru paste... i can see sceneTime is behind cloudflare... and does have reCaptcha
> 
> ![image](https://user-images.githubusercontent.com/597545/73726752-f63ac500-4794-11ea-80a2-cc34a99f1dd0.png)
> 
> where as earlier your output was that every provider had reCaptcha... could be weird bug from sickchill not detecting the raised exception.... anyway... 100% that would need to be reCaptcha solved before you could continue... @miigotu would need to add support for pass reCaptcha options for **cloudscraper**..

they were originally and i only pasted the worst run, not everytime with a new ip did it hit a cloudflare... only most of the time. good to know that even when working w/o cloudflare throwing a captcha that... theres still captchas...

**Comment by buccinator** [comment]
> Question... are you using tor exit nodes?

yes mate, i read that cloudflare is less strict if you use onion routing ?but then ill have to act as an exit node myself :- not sure?

**Comment by VeNoMouS** [comment]
> yes mate, i read that cloudflare is less strict if you use onion routing ?but then ill have to act as an exit node myself :- not sure?


![image](https://user-images.githubusercontent.com/597545/73728721-b544af80-4798-11ea-8d16-4e94962d91b8.png)

FWIW ... 99 times out of 100 you are gonna get blocked by cloudflare on tor exit nodes... least from my personal experience, you can test this even in a browser... just saying....

**Comment by buccinator** [comment]
> > yes mate, i read that cloudflare is less strict if you use onion routing ?but then ill have to act as an exit node myself :- not sure?
> 
> ![image](https://user-images.githubusercontent.com/597545/73728721-b544af80-4798-11ea-8d16-4e94962d91b8.png)
> 
> FWIW ... 99 times out of 100 you are gonna get blocked / reCaptcha'd by cloudflare on tor exit nodes... least from my personal experience, you can test this even in a browser... just saying....
> 
> @miigotu , not your code, not my code... its just how cloudflare is with tor....

yes, knew that and lead with that brother but as you pointed out some sites still use captchas anyway. if only i could use a PAC file on linux system terminal and set it for opensubtitles.org, all i really wanted back *cries*

still thought the reCaptcha error message was saying something meaningful ie. there was a parameter i could set to use a solver provider like @miigotu was saying earlier

**Comment by buccinator** [comment]
> Buy a VPN (or proxy service) subscription from a VPN provider and only route SickChill to the VPN-tunnel (or proxy) and not the other applications on that server.

i have only set it up for sickchill.. and couchpotato with the http_proxy and no_proxy env vars in the systemd script. there was only_proxy var only list instead of just no_proxy

**Comment by VeNoMouS** [comment]
yes, there is ....

change https://github.com/SickChill/SickChill/blob/master/sickbeard/helpers.py#L1256
to something like...

```python
        session = cloudscraper.create_scraper(
            recaptcha={
                'provider': 'anticaptcha',
                'api_key': 'your_anticaptcha_api_key'
           }
       )
```

**Comment by buccinator** [comment]
> yes, there is ....
> 
> change https://github.com/SickChill/SickChill/blob/master/sickbeard/helpers.py#L1256
> to something like...
> 
> ```python
>         session = cloudscraper.create_scraper(
>             recaptcha={
>                 'provider': 'anticaptcha',
>                 'api_key': 'your_anticaptcha_api_key'
>            }
>        )
> ```

ah legend.. cheers!!

**Comment by miigotu** [comment]
Some of those providers have an option to add your cookies in SC settings, which makes sure you get past the captcha without a solver

**Comment by miigotu** [comment]
Closing this for now, feel free to comment here if anything further comes up.

**Comment by Tiwas** [comment]
My error log is riddled with these...

**Comment by miigotu** [comment]
@Tiwas because you are on windows and the openssl version is too low. The last release for python2.7 on windows is linked to an old openssl that cant be updated.

**Comment by Tiwas** [comment]
@miigotu Thanks. Is Python 3 supported now? I seem to remember it wasn't always. If so I can upgrade.

**Comment by VeNoMouS** [comment]
FWIW...

Could be related... https://github.com/VeNoMouS/cloudscraper/issues/168 i'll be pushing out an update a bit just busy atm

**Comment by buccinator** [comment]
just a heads up if you guys follow through with the robot plugin. ive been behind a VPN since close of this thread and recently seen more of the same (no not tor, cyberghost) im sure its probably restricted to just one country and i could hop around. point being though eventually vpn nodes might one day be treated the same as tor anyway.
`2020-03-24 17:23:27 SEARCHQUEUE-BACKLOG-72614 :: [Torrentz] :: [f9161da] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. ()
2020-03-24 17:23:18 SEARCHQUEUE-BACKLOG-72614 :: [Torrentz] :: [f9161da] Request failed: !!Loop Protection!! We have tried to solve 3 time(s) in a row. ()
2020-03-24 17:21:30 SEARCHQUEUE-BACKLOG-72614 :: [LimeTorrents] :: [f9161da] Request failed: Cloudflare reCaptcha detected, unfortunately you haven't loaded an anti reCaptcha provider correctly via the 'recaptcha' parameter. ()
2020-03-24 17:21:24 SEARCHQUEUE-BACKLOG-72614 :: [LimeTorrents] :: [f9161da] Request failed: !!Loop Protection!! We have tried to solve 3 time(s) in a row. ()
2020-03-24 17:07:45 SEARCHQUEUE-MANUAL-377269 :: [Torrentz] :: [f9161da] Request failed: !!Loop Protection!! We have tried to solve 3 time(s) in a row. ()`

so far only two providers (on this node anyway)

thanks for all your work and help, was just checking back as the thread was active again
